src/ERC173/libraries/LibERC173.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.30;
+
+/// @title ERC-173 Contract Ownership
+library LibERC173 {
+    /// @dev This emits when ownership of a contract changes.
+    event OwnershipTransferred(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+
+    /// @notice Thrown when attempting to transfer ownership of a renounced contract.
+    error OwnableAlreadyRenounced();
+
+    bytes32 constant STORAGE_POSITION = keccak256("compose.erc173");
+
+    /// @custom:storage-location erc8042:compose.erc173
+    struct ERC173Storage {
+        address owner;
+    }
+
+    /// @notice Returns a pointer to the ERC-173 storage struct.
+    /// @dev Uses inline assembly to access the storage slot defined by STORAGE_POSITION.
+    /// @return s The ERC173Storage struct in storage.
+    function getStorage() internal pure returns (ERC173Storage storage s) {
+        bytes32 position = STORAGE_POSITION;
+        assembly {
+            s.slot := position
+        }
+    }
+
+    /// @notice Get the address of the owner
+    /// @return The address of the owner.
+    function owner() internal view returns (address) {
+        return getStorage().owner;
+    }
+
+    /// @notice Set the address of the new owner of the contract
+    /// @dev Set _newOwner to address(0) to renounce any ownership.
+    /// @param _newOwner The address of the new owner of the contract
+    function transferOwnership(address _newOwner) internal {
+        ERC173Storage storage s = getStorage();
+        if (s.owner == address(0)) revert OwnableAlreadyRenounced();
```

Comment:
The idea with this lib func is really for initialization, right? I'm curious to hear your thoughts on this. I lean toward lib funcs being permissive. I think disallowing this works for 99% of cases, but this won't be usable with unique functionality like:

```
function autoRenounce() {
    if (block.timestamp < TIMESTAMP_TO_RENOUNCE) {
        revert TooEarly();
    }
    LibOwnable.transferOwnership(address(0));
    // Doesn't need to be called by owner and avoids
    // (in a Gollum voice): "Must keep the precious ownership"
}
```